### PR TITLE
fix(types): use a minimal type for transformSearchClient

### PR DESCRIPTION
**Pull request by millotp**
The `transformSearchClient` type has change in the last release (https://github.com/algolia/docsearch/pull/2326) and is breaking in some situation.
To help with the migration, we can expose a new type that is the bare minimum needed by docsearch.

### CODE REVIEW

File: packages/docsearch-react/src/DocSearch.tsx [hunk 1]
```diff
@@ -16,17 +15,23 @@ export type DocSearchTranslations = Partial<{
   modal: ModalTranslations;
 }>;
 
+export type DocSearchTransformClient = {
```
Comment (shortcuts) [inline packages/docsearch-react/src/DocSearch.tsx:18]:
```suggestion
// The interface that describes the minimal implementation required for the algoliasearch client, when using the [`transformSearchClient`](https://docsearch.algolia.com/docs/api/#transformsearchclient) option
export type DocSearchTransformClient = {
```

wdyt?

we also need to update the definition here https://docsearch.algolia.com/docs/api/#transformsearchclient

**Review by shortcuts — COMMENTED** [review]
noice

File: packages/docsearch-react/src/DocSearch.tsx (hunk 1 above)
Comment (millotp) [reply packages/docsearch-react/src/DocSearch.tsx:18]: good point

**Review by shortcuts — APPROVED** [review]
gg!

**Pull request merged**